app/controllers/sessions_controller.rb

```diff
@@ -38,4 +38,33 @@ def validate_otp
       render nothing: true, status: :unauthorized
     end
   end
+
+  def validate_qrcode
+    signin_params = params
+      .permit(:account_id, :code)
+      .to_hash
+      .symbolize_keys
+
+    user = User.find_by!(account_id: signin_params.fetch(:account_id))
+    form = Signin::ValidateMobileTokenForm.new(
+      user:, code: signin_params.fetch(:code)
+    )
```

Comment:
In case of you accept my previous suggestion.

```suggestion
    user = User.find_by!(account_id:)
    form = Signin::ValidateMobileTokenForm.new(user:, code:)
```

Reply:
Done! 2f7fa38ae843a061dae32cb3634c374892fcb37a